### ISSUE TITLE
Configuration extension does not merge `ban` entries

### ISSUE THREAD

**Issue by pyrocat101** [palantir/tslint#1951]
### Bug Report

- __TSLint version__: `4.1.0`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
Foo.bar("test");
```

with `tslint.json` configuration:

```json
{
  "extends": "./base_tslint.json",
  "rules": {
    "ban": [true, ["Foo", "baz"]]
  }
}
```

content of `base_tslint.json`:

```json
{
  "rules": {
    "ban": [true, ["Foo", "bar"]]
  }
}
```

#### Actual behavior

Lint pass.

#### Expected behavior

Warn about usage of `Foo.bar`.

**Comment by JoshuaKGoldberg** [comment]
Subset of #3474. It's an interesting area!
